### ISSUE TITLE
The speaker audio control periodically set to minimum without any user action

### ISSUE THREAD

**Issue by ElLamparto** [qTox/qTox#3478]
Debian 32bit, qTox updated yesterday
Reproducible: Always. Just let it run for some time.

**Comment by antis81** [comment]
What do you mean by "periodically"?

**Comment by zetok** [comment]
> qTox updated yesterday

Doesn't mean much. Version info?

**Comment by ElLamparto** [comment]
@antis81, I mean "from time to time" - I've made no statistics. No call or other obvious event is required for the bug to appear. The speaker volume is set to zero.

@zetok,  qTox version v1.4.0-232-g4eed254

**Comment by ElLamparto** [comment]
Las night I did some testing. The speaker volume is set to zero ALWAYS after qTox start. I would look at reading/writting od the config file. Recently the font definition was added, maybe it messes up the config...

**Comment by antis81** [comment]
> Recently the font definition was added, maybe it messes up the config...

It definitely doesn't.

Volume is set by setting the master (listener) volume of the audio output device. The value is taken from settings and drops zero, when the device is invalid for some reason. So we need to know exactly, if…
1. …this happens, when opening settings.
2. …what device did you chose (default?).
3. …it does happen, when **not** opening settings and just make a call.

**Comment by ElLamparto** [comment]
Currently I cannot answer those questions. What I can say is this:
- The speaker volume is set to zero after qTox start and going to Settings,
- Once the volume is corrected, going to Settings and back, even many times, changes nothing (the volume rest correct).
- Once the volume is corrected, audio calls are ok.

The bug was introduced quite recently (since I do not install every and each update, I cannot say which version was first at fault).

Tested on two different machines (both with Debian).

**Comment by ElLamparto** [comment]
Ok, here is the qTox log:
- I started qTox,
- entered AV settings twice,
- shut qTox down.

[qtoxLog.txt](https://github.com/tux3/qTox/files/353971/qtoxLog.txt)

AV works in spite of the warnings.

**Comment by antis81** [comment]
@ElLamparto Can you compile qTox? If so, can you please check, if #3507 fixes your problem? For me, it works.

**Comment by ElLamparto** [comment]
@antis81, I have Qt, but never compiled qTox. I could try it next week, unless your modif is available in the repo before (in which case, please let me know). Thanks!
By the way, is there a new repo for Debian? If so, which one?

**Comment by antis81** [comment]
@ElLamparto There's no need to compile qTox by yourself unless you want to help with testing PR's or contribute code. I'm going to merge #3507 soon and it is going to fix settings initialization.

According to the following lines from the log file, your input device (microphone) cannot be opened:

``` text
…
[21:14:48.652] src/audio/audio.cpp:326 : Debug: Opening audio input "SPC 1000NC PC Camera Analog Mono"
[21:14:48.652] src/audio/audio.cpp:344 : Warning: Failed to initialize audio input device: "SPC 1000NC PC Camera Analog Mono"
[21:14:48.652] src/audio/audio.cpp:274 : Warning: Failed to subscribe to audio input device.
…
```

PR #3507 is not going to fix this particular problem!

I have another question:
If you open qTox settings and go to "A/V Settings" tab, do you see the name as above in "Input  device"? As an advice, you can try and switch to another input device (e.g. default device) to make it work. OTOH you state, that volume drops to 0, which can only be related to settings GUI (which is fixed with #3507) - otherwise that makes absolutely no sense.
